File: monthcalendar.py
#!/usr/bin/env python
"""Print out the list of dates in this month, as a markdown list"""
import datetime
import sys


offset = 0
if len(sys.argv) > 1:
    offset = int(sys.argv[1])
today = datetime.date.today()
tmp = datetime.date(today.year, today.month + offset, 1)
end = datetime.date(today.year, today.month + 1 + offset, 1)
while tmp < end:
    print(f"- `{tmp.day:2d}`: ")
    tmp += datetime.timedelta(days=1)

File: monthcalendar.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"
)

func main() {
	var offset int64 = 0
	var err error
	if len(os.Args) > 1 {
		offset, err = strconv.ParseInt(os.Args[1], 10, 64)
	}
	t := time.Now()
	offsetMonth := time.Month(int64(t.Month()) + offset)
	t = time.Date(t.Year(), offsetMonth, 1, 0, 0, 0, 0, t.Location())

	tmp := t
	delta_1d, err := time.ParseDuration("24h")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Failed to parse timedelta: %v", err)
		os.Exit(1)
	}
	for tmp.Month() == t.Month() {
		fmt.Printf("- %v-%02v-%02v %v ~\n",
			t.Year(),
			int(t.Month()),
			t.Day(),
			t.Weekday().String()[:3],
		)
		t = t.Add(delta_1d)
	}
}
